src/OrchardCore/OrchardCore.Media.Abstractions/Events/MediaContext.cs

```diff
@@ -0,0 +1,23 @@
+namespace OrchardCore.Media.Events
+{
+    /// <summary>
+    /// Represents a media context.
+    /// </summary>
+    public class MediaContext
+    {
+        /// <summary>
+        /// The path of the file for this media to be created in the file store.
+        /// </summary>
+        public string Path { get; set; }        
+        
+        /// <summary>
+        /// Any dynamic input data.
+        /// </summary>
+        public object Input { get; set; }
+
```

Comment:
No. Not necesary, and not how OC likes to do things.

Anyone needing to pass custom data from say a custom media controller (best example I can think of that someone might actually do) should inherit from `MediaContext` provide their own property, and cast back up inside their custom event handler.

e.g. 
```
public class CustomMediaContext : MediaContext
{
    public float X { get; set; }
}
```
and inside event handler
```
var custom = context as CustomMediaContext;
if (custom == null)
  return;
```